### ISSUE TITLE
Issues with blech_process.py

### ISSUE THREAD

**Issue by Mraymon5** [abuzarmahmood/blech_clust#180]
I'm running into 2 issues running this script; one is a warning that may or may not be a problem, and the other is an error that's stopping the code.

The warning is this:
```
/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/base.py:318: UserWarning: Trying to unpickle estimator StandardScaler from version 0.23.2 when using version 1.2.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
```

I don't know if this is an issue, but I did just re-run the requirements installation part of the pipeline, and I was also getting the warning before that, so that neither created nor resolved the issue.

The error is this:
```
Traceback (most recent call last):
  File "blech_process.py", line 154, in <module>
    classifier_handler.classify_waveforms(
  File "/home/ramartin/blech_clust/utils/blech_process_utils.py", line 595, in classify_waveforms
    clf_prob = self.pred_pipeline.predict_proba(slices)[:, 1]
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/pipeline.py", line 546, in predict_proba
    Xt = transform.transform(Xt)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/pipeline.py", line 658, in transform
    Xt = transform.transform(Xt)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/utils/_set_output.py", line 140, in wrapped
    data_to_wrap = f(self, X, *args, **kwargs)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/pipeline.py", line 1242, in transform
    Xs = Parallel(n_jobs=self.n_jobs)(
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/utils/parallel.py", line 63, in __call__
    return super().__call__(iterable_with_config)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/joblib/parallel.py", line 1048, in __call__
    if self.dispatch_one_batch(iterator):
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/joblib/parallel.py", line 864, in dispatch_one_batch
    self._dispatch(tasks)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/joblib/parallel.py", line 782, in _dispatch
    job = self._backend.apply_async(batch, callback=cb)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/joblib/_parallel_backends.py", line 208, in apply_async
    result = ImmediateResult(func)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/joblib/_parallel_backends.py", line 572, in __init__
    self.results = batch()
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/joblib/parallel.py", line 263, in __call__
    return [func(*args, **kwargs)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/joblib/parallel.py", line 263, in <listcomp>
    return [func(*args, **kwargs)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/utils/parallel.py", line 123, in __call__
    return self.function(*args, **kwargs)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/pipeline.py", line 876, in _transform_one
    res = transformer.transform(X)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/pipeline.py", line 658, in transform
    Xt = transform.transform(Xt)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/utils/_set_output.py", line 140, in wrapped
    data_to_wrap = f(self, X, *args, **kwargs)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/decomposition/_base.py", line 120, in transform
    X = self._validate_data(X, dtype=[np.float64, np.float32], reset=False)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/base.py", line 588, in _validate_data
    self._check_n_features(X, reset=reset)
  File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/sklearn/base.py", line 389, in _check_n_features
    raise ValueError(
ValueError: X has 50 features, but PCA is expecting 75 features as input.
```

**Comment by abuzarmahmood** [comment]
Warning shouldn't be a problem. It's not ideal but I've seen it before and it works fine.

The error is weird, and it seems like there is a mismatch at the feature calculation level. I think it might be related to #179.
Could you try deleting and reinstalling (https://github.com/abuzarmahmood/blech_clust?tab=readme-ov-file#setup) neuRecommend and let me know if you're still getting the error.

**Comment by Mraymon5** [comment]
I don't think it's connected to #179; I think I actually caused #179 accidentally while trying to work through this issue.

Anyway, I deleted my entire cloned repository of blech_clust, and reinstalled, and the error is the same.

Actually, it brought up another error that I had dealt with the first time I installed blech_clust:

File "/home/ramartin/anaconda3/envs/blech_clust/lib/python3.8/site-packages/xarray/core/dask_array_compat.py", line 60, in <module>
    sliding_window_view = da.lib.stride_tricks.sliding_window_view
AttributeError: module 'dask.array' has no attribute 'lib'

Which was fixed by running `conda update dask` in the terminal. I think that's what I did the first time also

**Comment by abuzarmahmood** [comment]
Could you delete and reinstall `neurecommend` (not blech_clust)?

**Comment by Mraymon5** [comment]
Okay, tried that, no change. I did just try running it on a different day, and I'm not getting that error. Because I'd messed around with some of the files, I also tried pulling fresh copies of that days data off my backup and running that instead, and I get the same error.

The only obvious thing I can see differentiating the files is that it looks like the problem child was accidentally recorded at 20,000hz instead of 30,000, but I did change that value in the params files, and tried re-running everything on a clean copy of the data, and get the same error.

**Comment by abuzarmahmood** [comment]
Ah, ok. That error makes sense now. The pipeline for the classifier currently only works with waveform snippets of a specific length. Reducing the sampling rate will shorten that length and break the pipeline.

I would turn of neurecommend use in your `blech_clust/params/waveform_classifier_params.json`...so it should look like:

```
{
   "use_neuRecommend":false,
   "use_classifier":true,
   "throw_out_noise":false,
   "min_suggestion_count" : 2000
}
```
And then re-run processing.

Let me know if that works

**Comment by Mraymon5** [comment]
I made that change to the params file, and got this warning:


=== Performing manual clustering ===

======================================
Classifier output not found, please run blech_run_process.sh with classifier.
======================================

**Comment by abuzarmahmood** [comment]
So I couldn't QUITE find why that check was added at that particular place. It was during the addition of auto-clustering and I think it was misplaced.
I've updated `blech_process.py` and `utils/blech_process_utils.py` so that presence of classifier data is only checked when `use_classifier` is turned on and we want to use `create_classifier_plots`

Try pulling the latest commit from the branch and re-running. Let me know if it works.
